### PR TITLE
Fix crashes on visualization

### PR DESCRIPTION
**Pull request by vandanavk**
## Description ##
Fixes https://github.com/apache/incubator-mxnet/issues/14245
Fixes https://github.com/apache/incubator-mxnet/issues/10416

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [MXNET-$JIRA_ID], where $JIRA_ID refers to the relevant [JIRA issue](https://issues.apache.org/jira/projects/MXNET/issues) created (except PRs with tiny changes)
- [x] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [x] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- Check the API doc at http://mxnet-ci-doc.s3-accelerate.dualstack.amazonaws.com/PR-$PR_ID/$BUILD_ID/index.html
- [x] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
- Add a check for the presence of kernel in attrs

## Comments ##
Test code for Pooling visualization crash (after fix):
```
import mxnet as mx
data=mx.sym.Variable('data')
symbol = mx.sym.Pooling(data = data, global_pool = True, pool_type = 'avg')
graph = mx.viz.plot_network(symbol)
graph
```
image

Test code for LeakyRelu visualization crash (after fix):
```
import mxnet as mx
data=mx.sym.Variable('data')
symbol = mx.sym.LeakyReLU(data=data)
graph = mx.viz.plot_network(symbol)
graph
```

image

Test after changing to str.format:
```
import mxnet as mx
data=mx.sym.Variable('data')
symbol = mx.sym.Convolution(data=data, kernel=(3,3), stride=(1,1), num_filter=3)
graph = mx.viz.plot_network(symbol)
graph
```
image
```
import mxnet as mx
data=mx.sym.Variable('data')
symbol = mx.sym.FullyConnected(data=data, num_hidden=3)
graph = mx.viz.plot_network(symbol)
graph
```
image

Change in print_summary tested with 
```
nosetests tests/python/unittest/test_viz.py:test_print_summary
.
----------------------------------------------------------------------
Ran 1 test in 0.015s

OK

```

### CODE REVIEW

**Comment by vandanavk** [comment]
@mxnet-label-bot add [Visualization, pr-awaiting-review]

**Comment by vandanavk** [comment]
Test for plot network has been disabled in the code

```
@unittest.skipIf(not graphviz_exists(), "Skip test_plot_network as Graphviz could not be imported")
def test_plot_network():
```

File: python/mxnet/visualization.py [hunk 1]
```diff
@@ -339,11 +339,17 @@ def looks_like_weight(name):
         elif op == "BatchNorm":
             attr["fillcolor"] = cm[3]
         elif op in ('Activation', 'LeakyReLU'):
-            label = r"%s\n%s" % (op, node["attrs"]["act_type"])
+            if op == "LeakyReLU":
```
Comment (szha) [inline python/mxnet/visualization.py]: if special handling is needed for each case, then the outer `elif op in ...` is not necessary.

File: python/mxnet/visualization.py [hunk 2]
```diff
@@ -339,11 +339,17 @@ def looks_like_weight(name):
...
+                attrs = node.get("attrs")
+                act_type = attrs.get("act_type", "Leaky") if attrs else "Leaky"
+            else:
+                act_type = node["attrs"]["act_type"]
+            label = r"%s\n%s" % (op, str(act_type))
```
Comment (szha) [inline python/mxnet/visualization.py]: this usage is being deprecated. use string.format instead.

**Comment by vandanavk** [comment]
All review comments addressed. Details of tests are in the PR description

**Comment by wkcn** [comment]
The PR has been merged. Thank you again! : )

**Pull request merged**